p2p/transport/webrtc/stream.go

```diff
@@ -167,65 +181,134 @@ func (s *stream) Reset() error {
 	return closeReadErr
 }
 
+func (s *stream) closeForShutdown(closeErr error) {
+	defer s.cleanup()
+
+	s.mx.Lock()
+	defer s.mx.Unlock()
+
+	s.closeForShutdownErr = closeErr
+	s.SetReadDeadline(time.Now().Add(-1 * time.Hour))
```

Comment:
This seems like a strange way to do this. It'll also be racy (e.g., if the user calls the same function to change the read deadline).

Do we need to change the read deadline? If this is the only way to do this, let's add something to `SetReadDeadline` so it can't be changed again by the user.

Reply:
This is the only way to unblock a waiting reader. Even closing the datachannel won't unblock it till 1RTT later when the peer replies with a close datachannel on its part. 

I've changed this to make SetReadDeadline a noop after the read half is closed so the user can't change this again.